overlord/devicestate/devicemgr.go

```diff
@@ -995,6 +995,28 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 
 var ErrUnsupportedAction = errors.New("unsupported action")
 
+func (m *DeviceManager) RequestReboot(systemLabel, mode string) error {
+	// most simple case: just reboot
+	if systemLabel == "" && mode == "" {
+		m.state.Lock()
+		defer m.state.Unlock()
+
+		logger.Noticef("restarting system")
+		m.state.RequestRestart(state.RestartSystemNow)
+		return nil
+	}
+
+	// no systemLabel means "current" so get the current system label
+	if systemLabel == "" {
+		systemMode := m.SystemMode()
+		currentSys, _ := currentSystemForMode(m.state, systemMode)
+		systemLabel = currentSys.System
+	}
+
+	// hand over to RequestSystemAction now
+	return m.RequestSystemAction(systemLabel, SystemAction{Mode: mode})
```

Comment:
this needs unit tests, also maybe we should have a common lower level helper instead of calling RequestSystemAction, that one will do quite a few things or invert the relation and call Reboot from RequestSystemAction

Reply:
I added a bunch of unit tests now.